border0/resource_socket.go

```diff
@@ -728,6 +735,10 @@ func getResourceSocketUpdate(sem sem.Semaphore) schema.UpdateContextFunc {
 			}
 		}
 
+		if !config.IsPrimary {
```

Comment:
as commented above, this could just be `cfg.Delay()`  without the conditional. And have all the logic inside it.